src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -44,28 +47,90 @@ public IEnumerable<DataSource> ListAllDataSources()
         this._defaultDataSourceName = Guid.NewGuid().ToString();
-        this._dataSourceNameToDataSource.Add(this._defaultDataSourceName, this.DataSource);
+
+        // we will set them up with default values
+        this._dataSourceNameToDataSource = new Dictionary<string, DataSource>
+        {
+            { this._defaultDataSourceName, this.DataSource }
+        };
 
         this._entityNameToDataSourceName = new Dictionary<string, string>();
+
         foreach (KeyValuePair<string, Entity> entity in Entities)
         {
             _entityNameToDataSourceName.TryAdd(entity.Key, this._defaultDataSourceName);
         }
 
+        // Multiple database scenario.
+        this.DataSourceFiles = DataSourceFiles;
+        IEnumerable<KeyValuePair<string, Entity>> allEntities = Entities.AsEnumerable();
+
+        if (DataSourceFiles is not null)
+        {
+            try
+            {
+                // Iterate through all the datasource files and load the config.
+                IFileSystem fileSystem = new FileSystem();
+                FileSystemRuntimeConfigLoader loader = new(fileSystem);
+
+                foreach (string dataSourceFile in DataSourceFiles.SourceFiles ?? Enumerable.Empty<string>())
+                {
+                    if (loader.TryLoadConfig(dataSourceFile, out RuntimeConfig? config))
+                    {
+                        this._dataSourceNameToDataSource = this._dataSourceNameToDataSource.Concat(config._dataSourceNameToDataSource).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        this._entityNameToDataSourceName = this._entityNameToDataSourceName.Concat(config._entityNameToDataSourceName).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        allEntities = allEntities.Concat(config.Entities.ToList());
+                    }
+                }
+
+                this.Entities = new RuntimeEntities(allEntities.ToDictionary(x => x.Key, x => x.Value));
+            }
+            catch (Exception e)
+            {
+                // Errors could include invalid sub file paths, duplicated entity names, etc.
+                throw new DataApiBuilderException($"Error while loading datasource files: {e.Message}", HttpStatusCode.BadRequest, DataApiBuilderException.SubStatusCodes.ConfigValidationError);
+            }
+        }
+
+    }
+
+    /// <summary>
+    /// Constructor for runtimeConfig.
+    /// This constructor is to be used when dynamically setting up the config as opposed to using a cli json file.
+    /// </summary>
+    /// <param name="Schema"></param>
+    /// <param name="DataSource"></param>
+    /// <param name="Runtime"></param>
+    /// <param name="Entities"></param>
+    /// <param name="DefaultDataSourceName"></param>
+    /// <param name="DataSourceNameToDataSource"></param>
+    /// <param name="EntityNameToDataSourceName"></param>
+    /// <param name="DataSourceFiles"></param>
```

Comment:
nit: Either populate or remove. For newly added dicts for multi-db support, would prefer having a param def, in case someone new references the code.

Reply:
fixed.